src/Microsoft.Azure.WebJobs.Host/Utility.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Host
+{
+    internal static class Utility
+    {
+        /// <summary>
+        /// Implements a configurable exponential backoff strategy.
+        /// </summary>
+        /// <param name="exponent">The backoff exponent. E.g. for backing off in a retry loop,
+        /// this would be the retry attempt count.</param>
+        /// <param name="cancellationToken">The <see cref="CancellationToken"/> to use.</param>
+        /// <param name="unit">The time unit for the backoff delay. Default is 1 second, producing a backoff sequence
+        /// of 2, 4, 8, 16, etc. seconds.</param>
+        /// <param name="min">The minimum delay.</param>
+        /// <param name="max">The maximum delay.</param>
+        /// <returns>A <see cref="Task"/> representing the computed backoff interval.</returns>
+        public static async Task DelayWithBackoffAsync(int exponent, CancellationToken cancellationToken, TimeSpan? unit = null, TimeSpan? min = null, TimeSpan? max = null)
```

Comment:
We already have a `RandomizedExponentialBackoffStrategy` -- can you use that existing logic here?

Reply:
Yes, I'll move to that and remove the new code I added for this. I had considered that previously but thought the core backoff stuff was tied too deeply with the TaskSeriesTimer stuff (I was wrong).